Yosemite/Yosemite/Model/Storage/OrderStatsV4Totals+ReadOnlyConvertible.swift

```diff
@@ -11,27 +11,27 @@ extension Storage.OrderStatsV4Totals: ReadOnlyConvertible {
     public func update(with statsTotals: Yosemite.OrderStatsV4Totals) {
         totalOrders = Int64(statsTotals.totalOrders)
         totalItemsSold = Int64(statsTotals.totalItemsSold)
-        grossRevenue = statsTotals.grossRevenue
-        couponDiscount = statsTotals.couponDiscount
+        grossRevenue = NSDecimalNumber(decimal: statsTotals.grossRevenue)
```

Comment:
**I think it's time we all decide on a "home" for type transformations.** In general we've been defining the Network Model classes to use `NSDecimalNumber` as well, then do any type transformations in the ViewModel, not in the `ReadOnlyConvertible` extension.

The reason being: a lot of numbers often get transformed from integers / doubles / etc. into strings anyway, so we keep most of the transformations there. Same for strings - they often get converted into usable dates. (Conversions to Int64 are about the only exception to the rule.)

It makes perfect sense to me that we'd do some of our conversions here. And if we're going to use this option, let's define the cases and commit to keeping it consistent everywhere.

Reply:
`NSDecimalNumber` might be an edge case in Core Data, normally `NSDecimalNumber` in Objective-C is bridged to `Decimal` in Swift. [This SO post](https://stackoverflow.com/questions/41148782/swift-3-decimal-nsdecimal-and-nsdecimalnumber) quoted a swift-users forum answer:

> Objective-C APIs which use `NSDecimalNumber` should be bridged to `Decimal`, just as `NSString` is bridged to `String`. That bridging may not work for category methods, though

that's probably why `NSDecimalNumber` has to be used for Core Data properties for `Decimal` (otherwise it crashes because it does not recognize the `Decimal` setter).

I'm always pro consistency, a p2 on our conversion convention sounds great!